xwiki-platform-core/xwiki-platform-display/xwiki-platform-display-macro/src/main/java/org/xwiki/rendering/macro/display/DisplayMacroParameters.java

```diff
@@ -66,6 +67,7 @@ public class DisplayMacroParameters
     @PropertyGroup("stringReference")
     @PropertyFeature("reference")
     @PropertyDisplayType(EntityReferenceString.class)
+    @PropertyMandatory
```

Comment:
You cannot do that, since it's not that simple. There is a reason why this was not done like this: there are several way to provide the reference like setting the page (or the document when including legacy API).

Reply:
ah right, I forgot this... good catch Thomas

Reply:
We should add a comment in the file to explain it.

Reply:
We would need a mandatory group annotation to say we need one of the listed option to be set.

Reply:
Yes, the right way to fix this is to add the concept of mandatory to @PropertyFeature.